app/src/main/java/com/expidev/gcmapp/MeasurementDetailsActivity.java

```diff
@@ -504,4 +476,142 @@ private LinearLayout createRow(String name, int value)
+            new NewDetailsPageRetrieverTask().execute(this, measurementId, ministryId, mcc, period);
+        }
+    }
+
+    private void sortSixMonthAmounts(MeasurementDetails measurementDetails)
+    {
+        Collections.sort(measurementDetails.getSixMonthTotalAmounts(), sixMonthAmountsComparator());
+        Collections.sort(measurementDetails.getSixMonthLocalAmounts(), sixMonthAmountsComparator());
+        Collections.sort(measurementDetails.getSixMonthPersonalAmounts(), sixMonthAmountsComparator());
+    }
+
+    private Comparator<SixMonthAmounts> sixMonthAmountsComparator()
+    {
+        return new Comparator<SixMonthAmounts>()
+        {
+            @Override
+            public int compare(SixMonthAmounts lhs, SixMonthAmounts rhs)
+            {
+                DateFormat dateFormat = new SimpleDateFormat("yyyy-MM", Locale.getDefault());
+
+                try
+                {
+                    Date leftDate = dateFormat.parse(lhs.getMonth());
+                    Date rightDate = dateFormat.parse(rhs.getMonth());
+                    return leftDate.compareTo(rightDate);
+                }
+                catch(ParseException e)
+                {
+                    return 0;
+                }
+            }
+        };
+    }
+
+    private void onLoadDetailsFromServer(MeasurementDetails measurementDetails)
+    {
+        if(measurementDetails != null)
+        {
+            handleRetrievedMeasurementDetails(measurementDetails);
+            MeasurementsService.saveMeasurementDetailsToDatabase(this, measurementDetails);
+        }
+        else
+        {
+            Log.w(TAG, "No measurement detail data");
+            finish();
+        }
+    }
+
+    private class MeasurementDetailsLoaderCallbacks extends SimpleLoaderCallbacks<MeasurementDetails>
+    {
+        @Override
+        public Loader<MeasurementDetails> onCreateLoader(final int id, @Nullable final Bundle args)
+        {
+            switch(id)
+            {
+                case LOADER_MEASUREMENT_DETAILS:
+                    return new MeasurementDetailsLoader(MeasurementDetailsActivity.this, args);
+                default:
+                    return null;
+            }
+        }
+
+        @Override
+        public void onLoadFinished(
+            @NonNull final Loader<MeasurementDetails> loader,
+            @Nullable final MeasurementDetails measurementDetails)
+        {
+            switch(loader.getId())
+            {
+                case LOADER_MEASUREMENT_DETAILS:
+                    onLoadMeasurementDetails(measurementDetails);
+                    break;
+            }
+        }
+    }
+
+    private class NewDetailsPageRetrieverTask extends AsyncTask<Object, Void, MeasurementDetails>
+    {
+        @Override
+        protected MeasurementDetails doInBackground(Object... params)
+        {
+            Context context = (Context) params[0];
```

Comment:
you actually don't have to pass the context object in via execute, it would be available directly through `MeasurementDetailsActivity.this` since this isn't a static class.